WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -393,17 +394,50 @@ public ICoinsView AsAllCoinsView()
 
 	public ICoinsView Available() => AsCoinsView().Available();
 
-	public ICoinsView ChildrenOf(SmartCoin coin) => AsCoinsView().ChildrenOf(coin);
-
 	public ICoinsView CoinJoinInProcess() => AsCoinsView().CoinJoinInProcess();
 
 	public ICoinsView Confirmed() => AsCoinsView().Confirmed();
 
-	public ICoinsView DescendantOf(SmartCoin coin) => AsCoinsView().DescendantOf(coin);
+	public ImmutableArray<SmartCoin> DescendantOfAndSelf(SmartCoin coin)
+	{
+		lock (Lock)
+		{
+			return DescendantOfAndSelfNoLock(coin);
+		}
+	}
+
+	/// <summary>Gets descendant coins of the given coin - i.e. all coins that spent the input coin, all coins that spent those coins, etc.</summary>
+	/// <remarks>Callers must acquire <see cref="Lock"/> before calling this method.</remarks>
+	private ImmutableArray<SmartCoin> DescendantOfAndSelfNoLock(SmartCoin coin)
+	{
+		HashSet<SmartCoin> allCoins = new(Coins);
+		allCoins.UnionWith(SpentCoins);
```

Comment:
`AsAllCoinsViewNoLock()` should be used here, I believe.

Reply:
I agree that it makes little sense; we can introduce it here and change it in other places? I believe that in classes like this one, consistency is really important so it's easier to understand.